fix(elasticsearch): Domain.fromDomainAttributes gives "Invalid URL" when endpoint is a token

HelioStrike opened the pull request:
`Domain.fromDomainAttributes` throws the error "Invalid URL" when a token endpoint is provided. In this PR, the domain name is retrieved from the scope using the arn present in `DomainAttributes`, and if no name is found, we try to get it from the endpoint (as it has been done so far). Also added a test to verify that the error disappears with this change, even when it is given token attributes.

closes #15188

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 191c4b581dbce3389d17a83d370b26b76fabccdd
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7f879d40-e91b-47d2-be8e-a4c9fbe5a141%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

BenChaimberg commented:
Please add a summary description of the changes contained in this PR

BenChaimberg commented on packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:
```diff
@@ -913,6 +913,15 @@ describe('import', () => {
+    const domainArn = new CfnParameter(stack, 'domainArn', { type: 'String' }).valueAsString;
+    const domainEndpoint = new CfnParameter(stack, 'domainEndpoint', { type: 'String' }).valueAsString;
+
+    expect(() => {
+      Domain.fromDomainAttributes(stack, 'Domain', { domainArn, domainEndpoint });
+    }).not.toThrowError(/Invalid URL/);
```
I would prefer if the verification in this test was similar to the previous tests (confirming name, ARN, and endpoint are concrete values)

HelioStrike replied:
Checked equality with the CfnParameters. I'm new to this repo, so not sure how to turn concrete values into Tokens and then use them.

BenChaimberg replied:
yep this is what I wanted!

BenChaimberg reviewed (changes requested):
the modes of these files were changed to allow execution; please revert

HelioStrike commented:
@BenChaimberg Updated the PR. Had to change modes due to some permission issue with yarn. Reverted it now, and also modified tests to check equality.
Also, when modifying a PR, should I force push modified commits, or add a commit on top?

HelioStrike commented:
@BenChaimberg Updated the PR. Had to change modes due to some permission issue with yarn. Reverted it now, and also modified tests to check equality.
Also, when modifying a PR, should I force push modified commits, or add a commit on top?

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 2b2182f994bd32bcd12446191d12d446121bf7fc
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f9533c17-d3d8-455c-a65e-431c9ff7d8c9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: d6aad5486534566b79fab392a6af3338e2e71108
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fb790657-7543-43a5-aab6-5fe1bb9a5d5e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: c67b10b3c66ff7c5e82348e62307dd0ffb60fe96
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bce561a1-f361-42aa-80c0-a05c908da3e3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

BenChaimberg commented:
Please commit on top, as Github doesn't have great support for diffing between force-pushed commits

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 10863154abd9ac558b1203cd3ff6b950fedb4dba
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fe48dc07-bab8-4b73-8b85-aff813c8194f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
